Auth0/WebViewProvider.swift

```diff
@@ -0,0 +1,163 @@
+//
+//  WebViewProvider.swift
+//  Auth0
+//
+//  Created by Desu Sai Venkat on 18/09/24.
+//  Copyright © 2024 Auth0. All rights reserved.
+//
+
+#if os(iOS)
+
+@preconcurrency import WebKit
+
+
+extension WebAuthentication {
+    static func webViewProvider(redirectionURL: URL, style: UIModalPresentationStyle = .fullScreen) -> WebAuthProvider {
+        return { url, callback  in
+            WebViewUserAgent(authorizeURL: url, redirectURL: redirectionURL, modalPresentationStyle: style, callback: callback)
+        }
+    }
+}
+
+extension WebViewUserAgent {
+    var topViewController: UIViewController? {
+        guard let root = UIApplication.shared()?.windows.last(where: \.isKeyWindow)?.rootViewController else {
+            return nil
+        }
+        return self.findTopViewController(from: root)
+    }
+    
+    private func findTopViewController(from root: UIViewController) -> UIViewController? {
+        if let presented = root.presentedViewController { return self.findTopViewController(from: presented) }
+        
+        switch root {
+        case let split as UISplitViewController:
+            guard let last = split.viewControllers.last else { return split }
+            return self.findTopViewController(from: last)
+        case let navigation as UINavigationController:
+            guard let top = navigation.topViewController else { return navigation }
+            return self.findTopViewController(from: top)
+        case let tab as UITabBarController:
+            guard let selected = tab.selectedViewController else { return tab }
+            return self.findTopViewController(from: selected)
+        default:
+            return root
+        }
+    }
+}
+
+class WebViewUserAgent: NSObject, WebAuthUserAgent {
+    
+    static let customSchemeRedirectionSuccessMessage = "com.auth0.webview.redirection_success"
+    static let customSchemeRedirectionFailureMessage = "com.auth0.webview.redirection_failure"
+    let defaultSchemesSupportedByWKWebview = ["http", "https"]
```

Comment:
I'd suggest checking with somebody from the security team if we should be supporting `http` schemes here (vs `https` only), e.g. when using an authorize URL override that proxies to Auth0.